pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -271,23 +271,25 @@ public String load(String instanceId) {
     }
     _lineageManager = lineageManager;
     _rebalancePreChecker = rebalancePreChecker;
-    _rebalancePreChecker.init(this, executorService);
+    _rebalancePreChecker.init(this, executorService, diskUtilizationThreshold);
   }
 
   public PinotHelixResourceManager(ControllerConf controllerConf, @Nullable ExecutorService executorService) {
     this(controllerConf.getZkStr(), controllerConf.getHelixClusterName(), controllerConf.getDataDir(),
         controllerConf.tenantIsolationEnabled(), controllerConf.getEnableBatchMessageMode(),
         controllerConf.getDeletedSegmentsRetentionInDays(), controllerConf.tieredSegmentAssignmentEnabled(),
         LineageManagerFactory.create(controllerConf),
-        RebalancePreCheckerFactory.create(controllerConf.getRebalancePreCheckerClass()), executorService);
+        RebalancePreCheckerFactory.create(controllerConf.getRebalancePreCheckerClass()), executorService,
+        controllerConf.getDiskUtilizationThreshold());
   }
 
   public PinotHelixResourceManager(ControllerConf controllerConf) {
     this(controllerConf.getZkStr(), controllerConf.getHelixClusterName(), controllerConf.getDataDir(),
         controllerConf.tenantIsolationEnabled(), controllerConf.getEnableBatchMessageMode(),
         controllerConf.getDeletedSegmentsRetentionInDays(), controllerConf.tieredSegmentAssignmentEnabled(),
         LineageManagerFactory.create(controllerConf),
-        RebalancePreCheckerFactory.create(controllerConf.getRebalancePreCheckerClass()), null);
+        RebalancePreCheckerFactory.create(controllerConf.getRebalancePreCheckerClass()), null,
+        controllerConf.getDiskUtilizationThreshold());
```

Comment:
Discussed with @J-HowHuang and @rajagopr and decided for rebalance it is better to use a lower threshold of say 0.9 for flagging purposes.

@J-HowHuang to add a new controller config for now for rebalance purposes and default it to 0.9: `controller.rebalance.disk.utilization.threshold` (feel free to choose a better name)